packages/jupyter-ai/jupyter_ai/chat_handlers/base.py

```diff
@@ -9,15 +9,39 @@
 from jupyter_ai.config_manager import ConfigManager, Logger
 from jupyter_ai.models import AgentChatMessage, HumanChatMessage
 from jupyter_ai_magics.providers import BaseProvider
+from traitlets.config import Configurable
 
 if TYPE_CHECKING:
     from jupyter_ai.handlers import RootChatHandler
 
 
-class BaseChatHandler:
+class BaseChatHandler(Configurable):
     """Base ChatHandler class containing shared methods and attributes used by
     multiple chat handler classes."""
 
+    # Class attributes
+    id: str = "base-chat-handler"
```

Comment:
Also, we don't need to suffix every chat handler ID with `-chat-handler` since that's part of the class name. `'base'` is sufficient here.